language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/contexts/AbstractDocumentServiceContext.java

```diff
@@ -132,47 +127,30 @@ public LSOperation operation() {
 
     @Override
     public List<Symbol> visibleSymbols(Position position) {
-        if (this.visibleSymbols == null) {
-            Optional<SemanticModel> semanticModel;
-            if (this.cancelChecker == null) {
-                semanticModel = this.workspaceManager.semanticModel(this.filePath);
-            } else {
-                semanticModel = this.workspaceManager.semanticModel(this.filePath, this.cancelChecker);
-            }
-            Optional<Document> srcFile = this.workspaceManager.document(filePath);
-
-            if (semanticModel.isEmpty() || srcFile.isEmpty()) {
-                return Collections.emptyList();
-            }
+        Optional<SemanticModel> semanticModel;
+        if (this.cancelChecker == null) {
+            semanticModel = this.workspaceManager.semanticModel(this.filePath);
+        } else {
+            semanticModel = this.workspaceManager.semanticModel(this.filePath, this.cancelChecker);
+        }
```

Comment:
We can write a one liner here if suitable

Reply:
with a ternary condition? `condition ? val1 : val2`

Reply:
IMO this way is better than that